pkg/replicatedapp/api.go

```diff
@@ -87,20 +92,29 @@ func getLicenseFromAPI(url string, licenseID string) (*LicenseData, error) {
 // GetApplicationMetadata will return any available application yaml from
 // the upstream. If there is no application.yaml, it will return
 // a placeholder one
-func GetApplicationMetadata(upstream *url.URL, versionLabel string) ([]byte, error) {
-	metadata, err := getApplicationMetadataFromHost("replicated.app", upstream, versionLabel)
+func GetApplicationMetadata(upstream *url.URL, versionLabel string) (*ApplicationMetadata, error) {
+	host := util.GetReplicatedAPIEndpoint()
+	manifest, err := getApplicationMetadataFromHost(host, "metadata", upstream, versionLabel)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get metadata from replicated.app")
+		return nil, errors.Wrapf(err, "failed to get metadata from %s", host)
 	}
 
-	if len(metadata) == 0 {
-		metadata = []byte(DefaultMetadata)
+	if len(manifest) == 0 {
+		manifest = []byte(DefaultMetadata)
+	}
+
+	branding, err := getApplicationMetadataFromHost(host, "branding", upstream, versionLabel)
```

Comment:
Should we unmarshal the manifest here and check if the branding fields are empty or not before making this request?  Could prevent unnecessary requests for branding.

Reply:
i think this should be ok